ocaml-lsp-server/src/hover_req.ml

```diff
@@ -148,36 +140,109 @@ let hover_at_cursor parsetree (`Logical (cursor_line, cursor_col)) =
     (if is_at_cursor mtyp.pmty_loc then
        match mtyp.pmty_desc with
        | Pmty_ident { loc; _ } ->
-         if is_at_cursor loc then result := Some `Type_enclosing
+         if is_at_cursor loc then result := Some Type_enclosing
        | _ -> ());
     Ast_iterator.default_iterator.module_type self mtyp
   in
   (* Hover structure items *)
   let structure_item (self : Ast_iterator.iterator)
       (item : Parsetree.structure_item) =
     match item.pstr_desc with
+    | Pstr_type (_, tds) ->
+      List.iter
+        ~f:(fun (td : Parsetree.type_declaration) ->
+          let deriving_attr =
+            List.find
+              td.ptype_attributes
+              ~f:(fun (attr : Parsetree.attribute) ->
+                is_at_cursor attr.attr_loc && is_deriving_attr attr)
+          in
+          match deriving_attr with
+          | Some attr -> result := Some (Ppx_str_attr (item, attr.attr_loc))
+          | None -> result := Some Type_enclosing)
```

Comment:
I think we should continue the iteration in the case were we did not find that the cursor is on a ppx attribute.

Reply:
I have continued the iteration for the other node types but there won't be any type enclosing information. so if the user hovers on the name of a type declaration, they won't see any type enclosing info